connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -582,10 +582,12 @@ private static JsonNode convertToJson(Schema schema, Object logicalValue) {
         if (logicalValue == null) {
             if (schema == null) // Any schema is valid and we don't have a default, so treat this as an optional schema
                 return null;
-            if (schema.defaultValue() != null)
-                return convertToJson(schema, schema.defaultValue());
+            // Value `null` is valid for an optional filed, even though the filed has a default value.
```

Comment:
```suggestion
            // Value `null` is valid for an optional field, even though the field has a default value.
```

Reply:
Thanks for your suggestion, I fixed the typo now.